### DISCUSSION TITLE
Добро пожаловать в NetSchoolApi Discussions!

### DISCUSSION THREAD

**Discussion by lentryd** [lentryd/NetSchoolApi#5] [Announcements]
## 👋 Добро пожаловать!
Мы используем Обсуждения как место для общения с другими членами нашего сообщества. Мы надеемся, что вы:
* Задаёте вопросы, которые вас интересуют.
* Делитесь идеями.
* Взаимодействуете с другими членами сообщества.
* Приветствуйте других и будьте непредубежденными. Помните, что это сообщество, которое мы
строим вместе 💪 .

Чтобы начать, прокомментируйте ниже, представив себя, и расскажите нам о том, что вы делаете с этим сообществом.

**Comment by nm17** [comment]
Всем привет

**Reply by lentryd** [reply]
✋

**Comment by TheEntropyShard** [comment]
Привет, не совсем понимаю как работает получение отчетов. Странный там механизм какой то, не подскажешь?

**Reply by lentryd** [reply]
Привет.
[Вот](https://github.com/lentryd/NetSchoolApi/blob/main/docs/examples/grades.md) получение отчёта об успеваемости по предмету

[Вот](https://github.com/lentryd/NetSchoolApi/blob/main/docs/examples/journal.md) получение отчёта об успеваемости и посещаемости

**Reply by TheEntropyShard** [reply]
Это не то, что мне нужно, я не понимаю как реализовано получение отчетов, вебсокеты какие то там.

**Reply by lentryd** [reply]
> ... вебсокеты какие то там. 

Это и вправду очень непонятный момент. Вообще всю логику я скопировал из клиента сетевого

**Reply by TheEntropyShard** [reply]
Ладно, спасибо, пишу свою библиотеку на Java, надеюсь разберусь

**Reply by lentryd** [reply]
В этом [чатике](https://t.me/netschoolapi) могут помочь.
Вроде даже кто-то уже писал версию для Java

**Reply by TheEntropyShard** [reply]
Благодарю

**Comment by lystoner** [comment]
Возможно ли использовать эту библиотеку на React Native?

**Reply by lentryd** [reply]
Привет.
Я не пробовал. Однако если там работают и другие функции/библиотеки NodeJS, то эта библиотека тоже будет работать

**Reply by lystoner** [reply]
просто я получаю такую ошибку 
`ERROR in ./node_modules/netschoolapi/dist/classes/Client/methods/url/join.js 6:15-30
Module not found: Error: Can't resolve 'path' in 'C:\Users\Tim\Documents\Sites\React Sites\messenger-app\node_modules\netschoolapi\dist\classes\Client\methods\url'`
Просто я не знаю, связана ли эта ошибка с библиотекой или нет

**Reply by lentryd** [reply]
А ты делаешь веб версию?

**Reply by lystoner** [reply]
сейчас да, хоть я и не думаю что это может вообще получиться из-за корс политики, сначала я пытался использовать эту библиотеку в react native, но там примерно такая же ошибка
Хотя сейчас переписав файлы конфига, судя по всему ошибка ушла, наверное сейчас попробую так же решить это в native

**Reply by lentryd** [reply]
Вообще библиотека заточена для NodeJS и в вебе может не работать из-за отсутствия некоторых модулей в js на браузерах и плюс проблемы с CORS
